pkg/arguments/arguments.go

```diff
@@ -185,6 +188,88 @@ func ParseKnownFlags(cmd *cobra.Command, argv []string, failOnUnknown bool) erro
 	return nil
 }
 
+// PreprocessUnknownFlagsWithId Parses known and unknown flags will take the command line arguments and map the ones
+// that fit with known flags.
+func PreprocessUnknownFlagsWithId(cmd *cobra.Command, argv []string) error {
```

Comment:
Can we get unit tests for this func, a lot going on here think it would help to have some unit tests on it :pray: